File: integration_tests/java/user/PreferencesDoNotTrackV1.java
package com.jsontypedef.jtdcodegendemo;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonValue;

import java.util.List;
import java.util.Map;


public class PreferencesDoNotTrackV1 extends PreferencesDoNotTrack {

  
  @JsonProperty("do_not_track")
  private PreferencesDoNotTrackV1DoNotTrack doNotTrack;

  
  @JsonProperty("opt_out_channels")
  private List<String> optOutChannels;


  
  public PreferencesDoNotTrackV1() {
  }
  


  public PreferencesDoNotTrackV1DoNotTrack getDoNotTrack() {
    return doNotTrack;
  }

  public void setDoNotTrack(PreferencesDoNotTrackV1DoNotTrack doNotTrack) {
    this.doNotTrack = doNotTrack;
  }

  public List<String> getOptOutChannels() {
    return optOutChannels;
  }

  public void setOptOutChannels(List<String> optOutChannels) {
    this.optOutChannels = optOutChannels;
  }

}

File: integration_tests/csharp/user/PreferencesDoNotTrackV1.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Jtd.JtdCodegenDemo
{
    /// <summary>
    /// Our post-GDPR do-not-track settings
    /// </summary>
    
    public class PreferencesDoNotTrackV1 : PreferencesDoNotTrack
    {

        [JsonProperty("version")]
        public const string Version = "v1";


        /// <value>
        /// A multi-level do-not-track setting
        /// </value>
        [JsonProperty("do_not_track")]
        public PreferencesDoNotTrackV1DoNotTrack DoNotTrack { get; set; }

        /// <value>
        /// Channels the user has opted out of tracking for.
        /// </value>
        [JsonProperty("opt_out_channels")]
        public IList<string> OptOutChannels { get; set; }





    }
}
